### PR TITLE
Bug fix for mtime anomalies in RadosGW and other places

### PR DESCRIPTION
**Pull request by adamemerson**

### CODE REVIEW

File: src/test/CMakeLists.txt [hunk 1]
```diff
@@ -1823,6 +1823,10 @@ target_link_libraries(test_cls_lock
   radostest
   )
 
+add_executable(test_time test_time.cc)
```
Comment (yehudasa) [inline src/test/CMakeLists.txt]: @adamemerson what about updating Makefile.am?

**Comment by cbodley** [comment]
I wonder if we can get the ceph-dencoder tests to check the encode/decode, instead of my hacked-up unit test. If we do want to keep test_time, we'll need to add it to the Makefile for automake as well.

**Comment by adamemerson** [comment]
I don't think so. We're instantiating template a template on std::chrono::timepoint so we can't add in a generate_test_instances method.

**Comment by cbodley** [comment]
@adamemerson I'm looking at src/test/encoding/types.h. I think we should at least add TESTs for ceph::real_time and ceph::mono_time. I'll add those, and see if it fails when I run without your fix.

**Comment by adamemerson** [comment]
You can't (or at least shouldn't be able to) encode/decode ceph::mono_time.

Since the epoch changes between machines/across reboots I added no way to
serialize it.

**Comment by cbodley** [comment]
@yehudasa Adam pushed an updated branch that adds the noinst_HEADERS part.

**Pull request merged**